hgtector/search.py

```diff
@@ -131,6 +131,14 @@ def __call__(self, args):
         # read and validate input data
         self.input_wf()
 
+	# check requested threads if set vs available ones.
+        m=len(sched_getaffinity(0))
+        if self.threads and self.threads > m:
+            print("WARNING threads limited to %d (requested %d)" %(m,self.threads), file=sys. stderr)
```

Comment:
Good idea! I like printing to stderr. However currently it isn't adopted elsewhere in the program. In the future I may consider doing `logging` instead. How about we do the following for now:

```python
print(f'WARNING: Threads limited to {m} (requested {self.threads}).')
````